File: part3/script/ruby/hello_server.rb
#!/usr/bin/env ruby

require 'thrift'
$:.push('gen-rb')
require 'hello_svc'

class HelloHandler
  def getMessage(name)
    return 'Hello ' + name
  end
end

port = 9095
handler = HelloHandler.new()
processor = HelloSvc::Processor.new(handler)
trans_svr_ep = Thrift::ServerSocket.new(port)
trans_buf_fac = Thrift::BufferedTransportFactory.new()
proto_fac = Thrift::BinaryProtocolFactory.new()
server = Thrift::SimpleServer.new(processor, trans_svr_ep, trans_buf_fac, proto_fac)

puts "Starting server on port #{port}..."
server.serve()

File: part3/script/perl/HelloServer.pl
#!/usr/bin/env perl

use lib './gen-perl';

use utf8;
use strict;
use warnings;

use Thrift::Socket;
use Thrift::Server;
use helloSvc;

package HelloSvcHandler {
    sub new {
        my ($class, %opts) = @_;
        bless {}, $class;
    }

    sub getMessage {
        my $self = shift;
        my $input = shift;

        return "Hello ".$input;
    }
}

my $trans  = Thrift::ServerSocket->new(9095);
my $proc   = helloSvcProcessor->new(HelloSvcHandler->new);
my $server = Thrift::ForkingServer->new($proc, $trans);

$server->serve;

